File: thursday/week8/hagyoung99/Q10866.java
package thursday.week8.hagyoung99;

import java.io.*;
import java.util.*;
public class Q10866 {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        StringTokenizer st;
        StringBuilder sb = new StringBuilder();

        Deque<Integer> d = new ArrayDeque<>();
        while(N-- > 0){
            st = new StringTokenizer(br.readLine());
            String command = st.nextToken();
            switch(command){
                case "push_front":
                    d.addFirst(Integer.parseInt(st.nextToken()));
                    break;
                case "push_back":
                    d.addLast(Integer.parseInt(st.nextToken()));
                    break;
                case "pop_front":
                    if(d.isEmpty()) sb.append("-1").append("\n");
                    else sb.append(d.pollFirst()).append("\n");
                    break;
                case "pop_back":
                    if(d.isEmpty()) sb.append("-1").append("\n");
                    else sb.append(d.pollLast()).append("\n");
                    break;
                case "size":
                    sb.append(d.size()).append("\n");
                    break;
                case "empty":
                    if(d.isEmpty()) sb.append(1).append("\n");
                    else sb.append(0).append("\n");
                    break;
                case "front":
                    if(d.isEmpty()) sb.append("-1").append("\n");
                    else sb.append(d.peekFirst()).append("\n");
                    break;
                case "back":
                    if(d.isEmpty()) sb.append("-1").append("\n");
                    else sb.append(d.peekLast()).append("\n");
                    break;
            }
        }
        System.out.println(sb);
    }
}

File: saturday/week_08/gayion/Q10866.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');
input.shift();

const deque = [];
let output = '';

input.forEach(input => {
  if (input.includes('push_front')) {
    //splice로 제일 앞에 추가한다.
    deque.splice(0, 0, input.split(' ')[1]);
  }
  if (input.includes('push_back')) {
    //push로 제일 뒤에 추가한다.
    deque.push(input.split(' ')[1]);
  }
  if (input === 'pop_front') {
    //shift로 앞에 있는 수를 제거 후 출력한다.
    output += deque.length ? deque.shift() + '\n' : '-1' + '\n';
  }
  if (input === 'pop_back') {
    //pop으로 뒤에 있는 수를 제거 후 출력한다.
    output += deque.length ? deque.pop() + '\n' : '-1' + '\n';
  }
  if (input === 'size') {
    //개수를 출력한다.
    output += deque.length + '\n';
  }
  if (input === 'empty') {
    output += deque.length ? 0 + '\n' : 1 + '\n'
  }
  if (input === 'front') {
    //0번째 수를 출력한다.
    output += deque.length ? deque[0] + '\n' : '-1' + '\n'
  }
  if (input === 'back') {
    //마지막 수를 출력한다.
    output += deque.length ? deque[deque.length - 1] + '\n' : '-1' + '\n'
  }
})

console.log(output);